modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlCapabilityExtractor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.capability.xml;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+import org.mule.extension.annotations.capability.Xml;
+import org.mule.extension.introspection.capability.XmlCapability;
+import org.mule.extension.introspection.declaration.Construct;
+import org.mule.extension.introspection.declaration.Declaration;
+import org.mule.extension.introspection.declaration.DeclarationConstruct;
+import org.mule.extension.introspection.declaration.HasCapabilities;
+import org.mule.module.extension.CapabilityExtractor;
+
+/**
+ * Implementation of {@link CapabilityExtractor}
+ * that verifies if the extension is annotated with {@link org.mule.extension.introspection.capability.XmlCapability}
```

Comment:
If this class checks that a extensions has the XmlCapability, why the SpringBundleResourceContributor is checking that the capability exists? If it does not exists, that is a programming error and should not be ignored